### PR TITLE
feat(guard): add GuardService

### PR DESCRIPTION
**Pull request by vderunov**
Hello,
Created a new guard branch.
Created GuardService, which should check whether the user is authorized and, depending on this, give him access to the pages or not. GuardService will protect our routes! :)
Added in login.component.html - error message also in login.component.ts in ngOnInit methed added - checking router params code.

thank u

### CODE REVIEW

File: src/app/shared/guard.service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export class GuardService implements CanActivate {
+  constructor(private authenticationService: AuthenticationService, private router: Router) {
+  }
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): boolean {
+    if (this.authenticationService.isAuthenticated) {
+      return true;
+    } else {
```
Comment (ksenya-ta) [inline src/app/shared/guard.service.ts]: if it's returned already for 1 case no need to keep else

File: src/app/shared/guard.service.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import {ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot} from '@angular/router';
+import {Injectable} from '@angular/core';
+import {AuthenticationService} from './services/user/authentication.service';
```
Comment (ksenya-ta) [inline src/app/shared/guard.service.ts]: keep spaces inside {}

File: src/app/shared/guard.service.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+@Injectable({providedIn: 'root'})
+export class GuardService implements CanActivate {
+  constructor(private authenticationService: AuthenticationService, private router: Router) {
+  }
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): boolean {
+    if (this.authenticationService.isAuthenticated) {
```
Comment (ksenya-ta) [inline src/app/shared/guard.service.ts]: this flag will be removed in other PR, so you'll need to use Auth state service to check it instead

File: src/app/login/login.component.ts [hunk 4]
```diff
@@ -12,14 +12,22 @@ import { Router } from '@angular/router';
...
   constructor(
     private formBuilder: FormBuilder,
     private authenticationService: AuthenticationService,
-    private router: Router
+    private router: Router,
+    private route: ActivatedRoute
   ) {}
 
   ngOnInit() {
+    this.route.queryParams.subscribe((params: Params) => {
```
Comment (ksenya-ta) [inline src/app/login/login.component.ts]: unsubscribe

File: src/app/shared/guard.service.ts (hunk 2 above)
Comment (vderunov) [reply src/app/shared/guard.service.ts]: Corrected

Screenshot 2019-11-02 at 12 56 25

File: src/app/shared/guard.service.ts (hunk 1 above)
Comment (vderunov) [reply src/app/shared/guard.service.ts]: removed "else"

File: src/app/shared/guard.service.ts (hunk 3 above)
Comment (vderunov) [reply src/app/shared/guard.service.ts]: Created a getStateValue method in AuthStateService which returns the value of the stream.

Screenshot 2019-11-04 at 17 23 11

File: src/app/login/login.component.ts (hunk 4 above)
Comment (vderunov) [reply src/app/login/login.component.ts]: Created a unsubscribe using takeUntil

**Pull request merged**